### PR TITLE
Added cover guis for pump/filter

### PR DESCRIPTION
**Pull request by moller21**
Added for pump/filter, to check if you like it before i work on it more.

Guess I reinvented the wheel some times, it's not fun trying to use the api when all you have to go on is fun_41561_1 :)

From what i understand I probably should've used Container instead, so you can see the inventory to set the fluid filter.

I also started to work on when you interract with covers, it's abit wierd on machines. But nothing is changed yet.

### CODE REVIEW

File: src/main/java/gregtech/common/covers/GT_Cover_Pump.java [hunk 1]
```diff
@@ -67,26 +87,6 @@ public int doCoverThings(byte aSide, byte aInputRedstone, int aCoverID, int aCov
         return aCoverVariable;
     }
 
-    public int onCoverScrewdriverclick(byte aSide, int aCoverID, int aCoverVariable, ICoverable aTileEntity, EntityPlayer aPlayer, float aX, float aY, float aZ) {
```
Comment (draknyte1) [inline src/main/java/gregtech/common/covers/GT_Cover_Pump.java:70]: I wouldn't remove the screwdriver behaviour.

Comment (mitchej123) [reply src/main/java/gregtech/common/covers/GT_Cover_Pump.java:70]: 100%.  Leave both in.

**Comment by moller21** [comment]
How should the GUI be opened then? And I don't see why you'd still want screwdriver behavior.

**Comment by draknyte1** [comment]
Soldering Iron?

Not everyone wants to waste time opening a GUI when a single click with a screwdriver is faster.

**Comment by mitchej123** [comment]
> How should the GUI be opened then? And I don't see why you'd still want screwdriver behavior.

Have right clicking without a tool open the GUI?

**Comment by moller21** [comment]
Hm, soldering iron is an idea. I think no tool has some conflicts on some covers.

**Comment by repo-alt** [comment]
Soldering iron toggles strong redstone on the given side

**Comment by moller21** [comment]
> Soldering iron toggles strong redstone on the given side
Ah, thats true. 

Don't think no tool is a good solution because as far as I can see, you:
- Always show the "wrench grid"
- Not being able to configure on hidden side 
- None of the above.

**Comment by Glease** [comment]
I don't think GUI is significantly worse than one screwdriver right click... Alk is probably thinking about one click to switch to import mode. Let's move it to bare hand right click. 

Here is a summary of my idea

1. Bare hand right click will take all functionality of current screwdriver right click
2. Screwdriver right click will be the GUI.
3. Soldering iron left unchanged

**Comment by mitchej123** [comment]
I'm a fan of @Glease 's suggestion.  (Although I'd be on board with swapping #1 and #2, such that right clicking brings up the UI, and the screwdriver keeps the current behavior).

**Comment by draknyte1** [comment]
Agree with @mitchej123

**Comment by Dream-Master** [comment]
@moller21 can you add what mitch, alk and glee8e suggest ?

**Comment by moller21** [comment]
If we're gonna go with the no tool way, how do we deal with configuring on hidden sides? Do we always show the grid? Or only bein able to configure if you click on the actual cover?

**Comment by mitchej123** [comment]
We could make it on shift right click; and have the grid visible when sneaking?  Or *shudder* make a GUI configurator tool, since there doesn't look to be one that isn't in use already?

**Comment by moller21** [comment]
Hard hammer sounds perfect.

**Comment by Dream-Master** [comment]
@mitchej123 @draknyte1 @Glease @repo-alt @bartimaeusnek can we merge this now ?

File: src/main/java/gregtech/api/metatileentity/BaseMetaPipeEntity.java [hunk 2]
```diff
@@ -804,15 +805,6 @@ protected boolean canAccessData() {
         return hasValidMetaTileEntity() && !isDead;
     }
 
-    public boolean shouldDisplayWrenchGrid(ItemStack heldItem, byte side) {
```
Comment (draknyte1) [inline src/main/java/gregtech/api/metatileentity/BaseMetaPipeEntity.java]: Can you not remove methods? 🤔

File: src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java [hunk 3]
```diff
@@ -1318,19 +1318,17 @@ public int getUpgradeCount() {
         return (mMuffler ? 1 : 0) + (mLockUpgrade ? 1 : 0) + (mSteamConverter ? 1 : 0) + mOtherUpgrades;
     }
 
-    public boolean shouldDisplayWrenchGrid(ItemStack heldItem, byte side) {
```
Comment (draknyte1) [inline src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java]: And here...

File: src/main/java/gregtech/common/GT_Client.java [hunk 4]
```diff
@@ -130,7 +130,7 @@ public GT_Client() {
         });
     }
 
-    private static void drawGrid(DrawBlockHighlightEvent aEvent) {
+    private static void drawGrid(DrawBlockHighlightEvent aEvent, boolean showCoverConnections) {
```
Comment (draknyte1) [inline src/main/java/gregtech/common/GT_Client.java:133]: Create a secondary function that passes a default bool to this. Who knows what relies on these functions existing as is?

Comment (repo-alt) [reply src/main/java/gregtech/common/GT_Client.java:133]: anyone relying on private functions deserves trouble

**Comment by moller21** [comment]
I broke the commit thing, couldn't figure out how to revert.

That shouldDisplayWrenchGrid was a function i made in the first commit.

I think I'll make a new pull request.

**Comment by draknyte1** [comment]
> 
> 
> I broke the commit thing, couldn't figure out how to revert.
> 
> That shouldDisplayWrenchGrid was a function i made in the first commit.
> 
> I think I'll make a new pull request.

Oh that's fine then.

File: src/main/java/gregtech/common/GT_Client.java (hunk 4 above)
Comment (draknyte1) [reply src/main/java/gregtech/common/GT_Client.java:133]: Not saying anyone does, but there's 0 harm to come of adding in a function with the original arguments. 🤷‍♀️

**Pull request closed without merging**